src/Sulu/Component/Content/Metadata/ItemMetadata.php

```diff
@@ -245,20 +245,33 @@ public function getTitle($locale)
     }
 
     /**
-     * Return the paramter with the given name.
+     * Return the parameter with the given name.
      *
      * @param string $name
      */
-    public function getParameter($name)
+    public function getParameter($name, bool $nullIfNotExists = false)
     {
-        if (!isset($this->parameters[$name])) {
+        $matchingParameters = \array_values(
+            \array_filter(
+                $this->parameters,
+                function($parameter) use ($name) {
+                    if (!\is_array($parameter)) {
+                        return false;
+                    }
+
+                    return $parameter['name'] === $name;
+                }
+            )
+        );
```

Comment:
Why is that necessary?

Reply:
`params` of properties look like this: `[['name' => 'foo', 'type' => 'string', 'value' => 'bar']]` ... But this method previously handled the `$this->parameters` array, as it would look like this: `['foo' => 'bar']` ... So I had to somehow combine that logic

Reply:
So are you saying this didn't work at all before? :see_no_evil:

Reply:
I think so 😄

Reply:
Well, then it's a good thing it has been properly fixed now :see_no_evil: I guess what has been written in the unit test before was not how the data is actually structured, that's the disadvantages of these kind of tests :see_no_evil: In order to avoid something like that in the future, this parameter is now also called in some of the functional tests, right?